test/inspector/inspector-helper.js

```diff
@@ -0,0 +1,406 @@
+    assert(this.childInstanceDone, 'Child instance died prematurely');
+    this.returnCode_ = code;
+    this.running_ = false;
+  });
+};
+
+Harness.prototype.addStderrFilter = function(regexp, callback) {
+  this.stderrFilters_.push((message) => {
+    if (message.match(regexp)) {
+      setTimeout(callback);
+      return true;
+    }
+  });
+};
+
+Harness.prototype.run_ = function() {
+  setTimeout(() => {
+    this.task_(() => {
+      this.task_ = this.task_.next_;
+      if (this.task_)
+        this.run_();
+    });
+  });
+};
+
+Harness.prototype.enqueue_ = function(task) {
+  if (!this.task_) {
+    this.task_ = task;
+    this.run_();
+  } else {
+    let chain = this.task_;
+    while (chain.next_)
+      chain = chain.next_;
+    chain.next_ = task;
+  }
+  return this;
+};
+
+Harness.prototype.testHttpResponse = function(path, check) {
+  return this.enqueue_((doneCallback) => {
+    checkHttpResponse(this.port, path, (response) => {
+      check.call(this, response);
+      doneCallback();
+    });
+  });
+};
+
+Harness.prototype.runFrontendSession = function(tests) {
+  return this.enqueue_((callback) => {
+    http.get({
+      port: this.port,
+      path: '/node',
+      headers: {
+        'Connection': 'Upgrade',
+        'Upgrade': 'websocket',
+        'Sec-WebSocket-Version': 13,
+        'Sec-WebSocket-Key': 'key=='
+      }
+    }).on('upgrade', (message, socket) => {
+      const session = new TestSession(socket, this);
+      if (!(tests instanceof Array))
+        tests = [tests];
+      function enqueue(tests) {
+        session.enqueue((sessionCb) => {
+          if (tests.length) {
+            tests[0](session);
+            session.enqueue((cb2) => {
+              enqueue(tests.slice(1));
+              cb2();
+            });
+          } else {
+            callback();
+          }
+          sessionCb();
+        });
+      }
+      enqueue(tests);
+    }).on('response', () => assert(false, 'Upgrade was not received'));
+  });
+};
+
+Harness.prototype.expectShutDown = function(errorCode) {
+  this.enqueue_((callback) => {
+    if (this.running_) {
+      const timeoutId =
+          setTimeout(() => assert(false, 'Have not terminated'),
+          TIMEOUT);
+      this.process_.on('exit', (code) => {
+        clearTimeout(timeoutId);
+        assert.strictEqual(errorCode, code);
+        callback();
+      });
+    } else {
+      assert.strictEqual(errorCode, this.returnCode_);
+      callback();
+    }
+  });
+};
+
+exports.startNodeForInspectorTest = function(callback) {
+  const child = spawn(process.execPath,
+      [ '--inspect', '--debug-brk', mainScript ]);
+
+  const timeoutId = setTimeout(() => {
+    assert(false, 'Child process did not start properly');
+  }, TIMEOUT * 4); // 1 min should be reasonable
+
+  child.stderr.once('data', (data) => {
+    clearTimeout(timeoutId);
+    const text = data.toString('utf8');
+    const match = text.match(/Debugger listening on port (\d+)/);
```

Comment:
can't it be possible that the `Debugger listening ...` text isn't present after only one `data` event? I think it could be split in multiple events.